src/details/ArborX_Ray.hpp

```diff
@@ -273,6 +274,55 @@ KOKKOS_INLINE_FUNCTION float overlapDistance(Ray const &ray,
   return (tmax - tmin);
 }
 
+// Möller–Trumbore intersection algorithm
+KOKKOS_INLINE_FUNCTION bool rayTriangleIntersect(Ray const &ray,
+                                                 Triangle const &triangle,
+                                                 float &t, float &u, float &v)
+{
+  auto ab = makeVector(triangle.a, triangle.b);
+  auto ac = makeVector(triangle.a, triangle.c);
+
+  auto p = crossProduct(ray.direction(), ac);
+  auto det = dotProduct(ab, p);
+
+  auto const epsilon = 0.0000001f;
+  // If the determinant is negative the triangle is back-facing.
+  // If the determinant is close to 0, the ray misses the triangle because they
+  // are in the same plane.
+  if (det < epsilon)
```

Comment:
I think I prefer the previous condition. This could lead to unexpected behavior for users.

Reply:
Fixed